api/routes/projects.route.js

```diff
@@ -0,0 +1,23 @@
+const express = require("express");
+const router = express.Router();
+const projectValidator = require("../middlewares/validators/project.validator");
+
+const {
+  getUserProjects,
+  addNewProject,
+  updateProject,
+  deleteProject,
+  getProjectWithTasks,
+  addTaskToProject,
+  deleteTaskFromProject,
+} = require("../controllers/projects.controller");
+
+router.get("/", getUserProjects);
+router.get("/:id", getProjectWithTasks);
+router.post("/", projectValidator, addNewProject);
+router.patch("/:id/new-task", addTaskToProject);
```

Comment:
I guess that task validator also should be used

Reply:
Here I am just adding id of the task to the project's tasks array. That is why may be we don 't need it here.